server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -468,26 +468,34 @@ public void createSnapshotV2(final CreateSnapshotRequest request, final ActionLi
         long pinnedTimestamp = System.currentTimeMillis();
         final String repositoryName = request.repository();
         final String snapshotName = indexNameExpressionResolver.resolveDateMathExpression(request.snapshot());
-        validate(repositoryName, snapshotName);
 
-        final SnapshotId snapshotId = new SnapshotId(snapshotName, UUIDs.randomBase64UUID()); // new UUID for the snapshot
         Repository repository = repositoriesService.repository(repositoryName);
+        repository.executeConsistentStateUpdate(repositoryData -> new ClusterStateUpdateTask(Priority.URGENT) {
```

Comment:
Why is priority `URGENT`? For non-v2 snapshots, we use `NORMAL` priority.

Reply:
This is v2 snapshots only .